namespaces/execin.go

```diff
@@ -68,8 +72,33 @@ func ExecIn(container *libcontainer.Config, state *libcontainer.State, userArgs
 		return -1, terr
 	}
 
-	// Enter cgroups.
-	if err := EnterCgroups(state, cmd.Process.Pid); err != nil {
+	if config.Cgroups == nil {
+		// Enter existing cgroups of the container
+		// In this case, we cant keep track of individual execin session to
+		// clean up orphan processes, so just try to remove the main process
+		// when it fail
+		if err := EnterCgroups(state, cmd.Process.Pid); err != nil {
+			return terminate(err)
+		}
+	} else {
+		parent := container.Cgroups.Name
```

Comment:
Why not `parent := filepath.Join(container.Cgroups.Parent, container.Cgroups.Name)` and get rid of the 'if' condition?